sys-fs/xfsprogs/xfsprogs-9999.ebuild

```diff
@@ -0,0 +1,200 @@
+		--disable-lto
+		# The default value causes double 'lib'
+		--localstatedir="${EPREFIX}/var"
+		--with-crond-dir="${EPREFIX}/etc/cron.d"
+		--with-systemd-unit-dir="$(systemd_get_systemunitdir)"
+		--with-udev-rule-dir="$(get_udevdir)/rules.d"
+		$(use_enable icu libicu)
+		$(use_enable nls gettext)
+		$(use_enable libedit editline)
+	)
+
+	econf "${myconf[@]}"
+}
+
+src_compile() {
+	# -j1 for:
+	# gmake[2]: *** No rule to make target '../libhandle/libhandle.la', needed by 'xfs_spaceman'.  Stop.
+	emake V=1 -j1
+}
+
+src_install() {
+	# XXX: There's a missing dep in the install-dev target, so split it
+	emake DIST_ROOT="${ED}" HAVE_ZIPPED_MANPAGES=false install
+	emake DIST_ROOT="${ED}" HAVE_ZIPPED_MANPAGES=false install-dev
+
+	# Not actually used but --localstatedir causes this empty dir
+	# to be installed.
+	rmdir "${ED}"/var/lib/xfsprogs "${ED}"/var/lib || die
+
+	if ! use static-libs; then
+		rm "${ED}/usr/$(get_libdir)/libhandle.a" || die
+	fi
+
+	find "${ED}" -name '*.la' -delete || die
+}
+
+pkg_postrm() {
+	udev_reload
+}
+
+pkg_postinst() {
+	udev_reload
+}
+
+prepare_release() { # updates the version number in several places
+
+	# pass it to sed so it returns 1 when no replace occurred
+	local RS=';tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x'
+
+	parse_version() {
+		local -n parsed_version="${1}" # modify the variable passed as reference
+		parsed_version="$(grep -o -- \
+			'^AC_INIT(\[xfsprogs\],\[[0-9]\+\.[0-9]\+\.[0-9]\+\],\[linux-xfs@vger.kernel.org\])$' \
+			configure.ac || die)"
+
+		parsed_version=$( echo "${parsed_version}" | ( sed -r \
+			's/^AC_INIT\(\[xfsprogs\],\[([0-9]+).([0-9]+).([0-9]+)\],\[linux-xfs@vger\.kernel\.org\]\)$/\1.\2.9999/'"${RS}" \
+			|| die ) )
+	}
+
+	isolate_updateversion_function() {
+		# parses a function block from the upstream script
+		local updateversion_function
+		updateversion_function=$( sed -n \
+			'/^update_version() {/,/^}$/p' release.sh || die )
+		updateversion_function=$( echo "${updateversion_function}" \
+			| ( sed -r 's/^update_version\(\) \{$//'"${RS}" || die ) )
+		updateversion_function=$( echo "${updateversion_function}" \
+			| ( sed -r 's/^\}$//'"${RS}" || die ) )
+		echo "${updateversion_function}" > ./release_update_version.sh
+		chmod +x release_update_version.sh
+	}
+
+	update_VERSION_file() {
+		local version="$1"
+		local version_file
+		local version_major
+		local version_minor
+		local version_revision
+
+		version_major=$( echo "${version}" | ( sed -r \
+			's/^([0-9]+).([0-9]+).([0-9]+)$/\1/'"${RS}" || die ) )
+		version_file+='PKG_MAJOR='"${version_major}"$'\n'
+
+		version_minor=$( echo "${version}" | ( sed -r \
+			's/^([0-9]+).([0-9]+).([0-9]+)$/\2/'"${RS}" || die ) )
+		version_file+='PKG_MINOR='"${version_minor}"$'\n'
+
+		local version_revision=$( echo "${version}" | ( sed -r \
+			's/^([0-9]+).([0-9]+).([0-9]+)$/\3/'"${RS}" || die ) )
+		version_file+='PKG_REVISION='"${version_revision}"$'\n'
+
+		echo "${version_file}" > ./VERSION
```

Comment:
You should set all these variables first, then do:

```bash
		cat <<- EOF >> ./VERSION || die
			PKG_MAJOR=${version_major}
			PKG_MINOR=${version_minor}
			PKG_REVISION=${version_revision}
		EOF
```

It is *much* more readable. It also writes the file *once*, and correctly handles `die` for the file write.